admin_ui/app/modals/permissionModalsStore.js

```diff
@@ -0,0 +1,261 @@
+import { createAction, handleActions } from 'redux-actions';
+import noop from 'lodash/noop';
+
+// initial state of the modal store
+export const initialState = {
+  // A flag to establish the currently showing modal
+  // Possible values: ['PERMISSION_CREATE', 'PERMISSION_DELETE']
+  displayedModal: '',
+  // Permission-related entries
+  permission: {},
+  permissionCreateError: '',
+  permissionEditError: '',
+  permissionDeleteError: '',
+  isPermissionCreatePending: false,
+  isPermissionEditPending: false,
+  isPermissionDeletePending: false,
+};
+
+export const callbacks = {
+  onPermissionCreateSubmit: noop,
+  onPermissionCreateCancel: noop,
+  onPermissionEditSubmit: noop,
+  onPermissionEditCancel: noop,
+  onPermissionDeleteSubmit: noop,
+  onPermissionDeleteCancel: noop,
+};
+
+// ACTIONS
+
+// ****************************************************************************
+// Permission CREATE
+// ****************************************************************************
+export const openPermissionCreateModal = createAction(
+  'PERMISSION_CREATE_MODAL_OPEN',
+  (onSubmit = noop, onCancel = noop) => {
```

Comment:
Arguments as function params make it harder to refactor in the future (e.g. add yet another argument) - whereas passing arguments as object props makes the API more flexible. I prefer the latter, although I have no strong opinion.

Reply:
I think we blindly followed the structure for a `payloadCreator` method as specified here: https://redux-actions.js.org/api/createaction. Or am I missing something on this comment?